### PR TITLE
[wasm] Implement MINT_SWITCH opcode in jiterpreter

### PR DESCRIPTION
**Pull request by kg**
This PR implements basic support for switches in the jiterpreter, with a configurable size limit (big ones would eat up all the available trace space) that can be set to 0 in order to disable them.

Switch targets that go backwards (Vlad says these should be exceedingly rare) will cause a runtime bailout, as will switches that are too big. In local testing these bailouts are very rare, even with the current low threshold.

If a switch targets places outside of the trace, only those switch targets that are outside will cause bailouts - the rest of them will still work. So even for large traces where half the switch arms aren't reachable, this is still an improvement over how it was before, I think.

In the future if we raise the max trace size we can also raise the size limit on switches - from looking at instrumentation, a limit of 64 or so is probably ideal.

This PR also fixes a bug I noticed - if the jiterpreter trace table filled up we were still generating+compiling a trace and then failing to put it into the table, which wasted CPU time.

### CODE REVIEW

File: src/mono/browser/runtime/jiterpreter-trace-generator.ts [hunk 1]
```diff
@@ -183,6 +183,34 @@ function getOpcodeLengthU16 (ip: MintOpcodePtr, opcode: MintOpcode) {
     }
 }
 
+function decodeSwitch (ip: MintOpcodePtr) : MintOpcodePtr[] {
+    mono_assert(getU16(ip) === MintOpcode.MINT_SWITCH, "decodeSwitch called on a non-switch");
+    const n = getArgU32(ip, 2);
+    const result = [];
+    /*
```
Comment (pavelsavara) [inline src/mono/browser/runtime/jiterpreter-trace-generator.ts:190]: please remove dead code

File: src/mono/browser/runtime/jiterpreter-support.ts [hunk 2]
```diff
@@ -1278,6 +1285,23 @@ class Cfg {
         }
     }
 
+    // It's your responsibility to wrap this in a block and follow it with a bailout!
```
Comment (pavelsavara) [inline src/mono/browser/runtime/jiterpreter-support.ts:1288]: It's caller's responsibility

**Review by pavelsavara — APPROVED** [review]
@BrzVlad could you please review the logic please ?

**Review by BrzVlad — APPROVED** [review]

File: src/mono/browser/runtime/jiterpreter-trace-generator.ts (hunk 1 above)
Comment (kg) [reply src/mono/browser/runtime/jiterpreter-trace-generator.ts:190]: It's not dead code

**Pull request merged**